### ISSUE TITLE
Chaingun no longer plays pickup sound or flashes screen

### ISSUE THREAD

**Issue by Blzut3** [ECWolfEngine/ECWolf#72]
**[Original report](https://bitbucket.org/ecwolf/ecwolf/issues/72) by Belmondo (Bitbucket: [DenisBelmondo](https://bitbucket.org/DenisBelmondo), GitHub: [DenisBelmondo](https://github.com/DenisBelmondo)).**

----------------------------------------

Exactly what the title says. Upon picking up the chaingun, the little riff that would normally play, well, doesn't. And the screen doesn't seem to flash either. Furthermore, BJ's face doesn't smile maniacally and the default chaingun seems to replace the MG42 in the techdemo.

EDIT: Hm, it seems that the jingle did play, the screen flashed and BJ smiled when I picked it up during my last testrun which was just a few seconds ago, but the techdemo problem persists. Changing title.

EDIT 2: Ah yes, this morning I found that the techdemo replaces the actor GatlingGun instead of GatlingGunUpgrade. That solves the weapon replacement issue, however the screenflash/sound problem still exists. Changing title back.

**Comment by Blzut3** [comment]
Should I be assuming you're running a development build?  The pickup seems to work normally, but I have no sound at the moment.  It looks like it might be bugged on subsequent pick ups though.

As for the tech demo, it probably needs to be upgraded for 1.2.

**Comment by Blzut3** [comment]
**Original comment by Belmondo (Bitbucket: [DenisBelmondo](https://bitbucket.org/DenisBelmondo), GitHub: [DenisBelmondo](https://github.com/DenisBelmondo)).**

----------------------------------------

Ah yes I forgot to state that I am using the latest Mercurial build. Alrighty then.

**Comment by Blzut3** [comment]
I've looked into this a bit more.  ECWolf definitely has a bug with the chaingun pickup not playing the sound or flashing after the first pickup.  Judging with wolf4sdl it looks like it should play the sound and flash, but not grin if you already have the chaingun.  (Please correct me if I'm wrong here.)

I've also tested the WeaponGiver behavior against ZDoom and the techdemo definitely needs to be upgraded for 1.2.  WeaponGiver gives the weapon specified and not the replacement.

**Comment by Blzut3** [comment]
**Original comment by Belmondo (Bitbucket: [DenisBelmondo](https://bitbucket.org/DenisBelmondo), GitHub: [DenisBelmondo](https://github.com/DenisBelmondo)).**

----------------------------------------

I popped into Wolf4SDL v1.7 and I gave myself a Chaingun via ILM and picked one up subsequently after. It appears that the screen flashes, the sound plays, and BJ grins. The DOS version seems to yield the same results after doing the above.
